src/main/java/core/basesyntax/service/serviceimpl/FruitReportCreateImpl.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax.service.serviceimpl;
+
+import core.basesyntax.exception.NegativeBalanceException;
+import core.basesyntax.service.interfaces.FruitReportCreate;
+import java.util.HashMap;
+import java.util.Map;
+
+public class FruitReportCreateImpl implements FruitReportCreate {
+    private static final String SEMICOL = ",";
+
+    public String createReport(HashMap<String, Integer> fruitQuantity) {
+        StringBuilder builder = new StringBuilder();
+        builder.append("fruit").append(SEMICOL).append("quantity").append(System.lineSeparator());
+        for (Map.Entry<String, Integer> fruits: fruitQuantity.entrySet()) {
+            if (fruits.getValue() < 0) {
```

Comment:
we should prevent this situation in the `PurchaseOperationHandler`, by throwing an exception there, where it is closer to the reason why it happened, on the attempt to buy more fruits than we have available in the storage